### PR TITLE
Masq 99 check username exist

### PR DESCRIPTION
**Pull request by G-Ray**

### CODE REVIEW

File: src/containers/Settings/Settings.jsx [hunk 1]
```diff
@@ -40,16 +41,29 @@ class Settings extends React.Component {
...
 
+  getUsernameLabel () {
+    const username = this.state['username']
+
+    if (this.isValid('username')) {
+      return 'Pseudo (affiché)'
+    }
+
+    return isUsernameAlreadyTaken(username)
+      ? 'Pseudo déjà utilisé. Veuillez en choisir un autre.'
```
Comment (G-Ray) [inline src/containers/Settings/Settings.jsx]: all text, and eveything displayed on the app may need to be reviewed before going to prod, by us and others, what do you think ?

Comment (leventdem) [reply src/containers/Settings/Settings.jsx]: yes, it can be done by Maps team also

**Review by leventdem — APPROVED** [review]
ok

File: src/containers/Settings/Settings.jsx [hunk 2]
```diff
@@ -40,16 +41,29 @@ class Settings extends React.Component {
...
   }
 
+  getUsernameLabel () {
+    const username = this.state['username']
+
+    if (this.isValid('username')) {
+      return 'Pseudo (affiché)'
+    }
+
+    return isUsernameAlreadyTaken(username)
```
Comment (sujeebant) [inline src/containers/Settings/Settings.jsx]: Bug if isValid('username') then return would be called without checking if username already exists ?
Maybe replace with something like the following :
```
if (! this.isValid('username')) {
    return 'Le pseudo ne doit pas contenir d\'espaces, et peut contenir les caractères spéciaux suivants: !?$#@()-*'
}

if (isUsernameAlreadyTaken(username)) {
    return 'Pseudo déjà utilisé. Veuillez en choisir un autre.'
}

return 'Pseudo (affiché)'
```

File: src/modals/Signup/Signup.jsx [hunk 3]
```diff
@@ -44,13 +45,25 @@ class Signup extends React.Component {
...
       case 'lastname': return isName(field)
-      case 'username': return isUsername(field)
+      case 'username': return isUsername(field) && !isUsernameAlreadyTaken(field)
       case 'password': return isPassword(field)
       case 'passwordConfirmation': return field === this.state.password
       default: return false
     }
   }
 
+  getUsernameLabel () {
```
Comment (sujeebant) [inline src/modals/Signup/Signup.jsx]: same as the getUsernameLabel in Login form

**Review by sujeebant — APPROVED** [review]

**Pull request merged**